src/main/java/io/dgraph/client/GrpcDgraphResult.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2016 DGraph Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.dgraph.client;
+
+import java.util.Stack;
+
+import com.google.common.base.Strings;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+
+/**
+ * Implementation of DgraphResult using Grpcresponse.Response.
+ *
+ * @author Edgar Rodriguez-Diaz
+ * @version 0.0.1
+ */
+public class GrpcDgraphResult extends DgraphResult<Graphresponse.Response> {
+
+    private GrpcDgraphResult(final Graphresponse.Response theRootResult) {
+        super(theRootResult);
+    }
+
+    @Override
+    public JsonObject toJsonObject() {
+        final Stack<Graphresponse.Node> nodes = new Stack<>();
+        final Stack<JsonObject> jsonNodes = new Stack<>();
+
+        final Graphresponse.Node rootNode = getResponse().getN();
+        final JsonObject rootJson = nodeToJson(rootNode);
+
+        nodes.add(rootNode);
```

Comment:
those methods are identical, but I guess for the context of the stack `push` should be more clear, I can change that.